t/lib/WTSI/NPG/HTS/TreePublisherTest.pm

```diff
@@ -107,6 +107,44 @@ sub publish_tree : Test(58) {
   check_metadata($irods, map { catfile($irods_tmp_coll, $_) } @observed_paths);
 }
 
+sub npg_publish_tree_pl_metadata_from_stdin : Test(3) {
+  my $source_path = "${data_path}/treepublisher";
+  my @attributes;  
+  push @attributes, {attribute => 'attr1', value => 'val1', units => q[]};
+  push @attributes, {attribute => 'attr2', value => 'val2', units => q[]};
+
+  my $metadata_text = JSON->new->utf8->encode(\@attributes);
+  my $metadata_file_in = "metadata.json.in";
```

Comment:
If you need to create a temporary file during tests and delete it afterwards, better to use
`File::Temp`.